st2common/st2common/content/loader.py

```diff
@@ -15,59 +15,104 @@
 
 import os
 
+import six
+from yaml.parser import ParserError
+
 from st2common.constants.meta import (ALLOWED_EXTS, PARSER_FUNCS)
 from st2common import log as logging
-from yaml.parser import ParserError
+
+__all__ = [
+    'ContentPackLoader',
+    'MetaLoader'
+]
 
 LOG = logging.getLogger(__name__)
 
 
 class ContentPackLoader(object):
-    def __init__(self):
-        self._allowed_content_types = ['sensors', 'actions', 'rules']
+    ALLOWED_CONTENT_TYPES = ['sensors', 'actions', 'rules']
+
+    def get_content(self, base_dirs, content_type):
+        """
+        Retrieve content from the provided directories.
+
+        Provided directories are searched from left to right. If a pack with the same name exists
+        in multiple directories, first pack which is found wins.
+
+        :param base_dirs: Directories to look into.
+        :type base_dirs: ``list``
 
-    def get_content(self, base_dir=None, content_type=None):
-        if content_type is None:
-            raise Exception('Content type unknown.')
+        :param content_type: Content type to look for (sensors, actions, rules).
+        :type content_type: ``str``
 
-        if not os.path.isdir(base_dir):
-            raise Exception('Directory containing content-packs must be provided.')
+        :rtype: ``dict``
+        """
+        assert(isinstance(base_dirs, list))
 
-        if content_type not in self._allowed_content_types:
-            raise Exception('Unknown content type %s.' % content_type)
+        if content_type not in self.ALLOWED_CONTENT_TYPES:
+            raise ValueError('Unsupported content_type: %s' % (content_type))
+
+        content = {}
+        pack_to_dir_map = {}
+        for base_dir in base_dirs:
```

Comment:
If I understand this right it seems that we will support fragmented packs e.g.

base_path_1/my_pack/actions
base_path_2/my_pack/sensor

In this case actions and sensor will be registered as belonging to the same pack but actually be from different base paths - right? If so is this intentional?